src/cpp/fastdds/subscriber/DataReaderImpl/SampleLoanManager.hpp

```diff
@@ -131,9 +131,7 @@ struct SampleLoanManager
         // Increment references of input payload
         CacheChange_t tmp;
         tmp.copy_not_memcpy(change);
-        item->owner = change->payload_owner();
-        change->payload_owner()->get_payload(change->serializedPayload, item->owner, tmp);
-        item->owner = tmp.payload_owner();
+        change->payload_owner()->get_payload(change->serializedPayload, tmp.serializedPayload);
         item->payload = tmp.serializedPayload;
         tmp.payload_owner(nullptr);
         tmp.serializedPayload.data = nullptr;
```

Comment:
We can now avoid using a temporary `CacheChange_t` here, and use `change->payload_owner()->get_payload(change->serializedPayload, item->payload);`